system/security/securesocket/securesocket.cpp

```diff
@@ -164,28 +164,31 @@ class CSecureSocket : implements ISecureSocket, public CInterface
     virtual void   read(void* buf, size32_t size)
     {
         size32_t size_read;
-        readTimeout(buf, size, size, size_read, 0, false);
+        // MCK - this was:
+        // readTimeout(buf, size, size, size_read, 0, false);
+        // but that is essentially a non-blocking read() and we want a blocking read() ...
+        readTimeout(buf, 0, size, size_read, WAIT_FOREVER, false);
```

Comment:
ESP support for HTTPS is certainly one potential area.  SOAPCALL another.

But one thing you could do is search the code for anywhere this method is called on a secure socket.  If it's used we can figure out a way to test.  If this particular method isn't used someplace securesocket is then testing that probably isn't necessary.